whois-nrtm4/src/main/java/net/ripe/db/nrtm4/dao/NrtmVersionInfoRepository.java

```diff
@@ -57,7 +58,7 @@ public NrtmVersionInfoRepository(
     public Optional<NrtmVersionInfo> findLastVersion(final NrtmSource source) {
         try {
```

Comment:
If we are building initial snapshot files  for all sources at once, just checking if this table is empty or not is suffice , there is no need to join with sources table and get last record (last record could be any source)? If we want to do this way , then we need to check if lastVersion is  available for all sources and build for the sources for which there is no initial version available but we cannot do that so best to check just for empty or not?

Reply:
actually this method is already unused except by tests. removed

Reply:
But this being uses in createSnapshot to check if we need to initialize the snapshot or not

Reply:
separated now into checking for sources and checking for versions with a source. no constraints or check on whether number of versions == number of sources, maybe should be added?